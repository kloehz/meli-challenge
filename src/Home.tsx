import './styles/main.scss'

export const Home = () => {

    return <h1>Funcionando</h1>
}
