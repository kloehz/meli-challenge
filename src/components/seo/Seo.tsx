import { Helmet } from "react-helmet"

interface ITitle {
    title?: string
}

export const Seo = ({title}: ITitle) => {


    return (
        <Helmet>
            <title>{`${title} | Mercado Libre`}</title>
            {title !== 'Inicio'
                ? <meta
                    name="description"
                    content={`Encontra ${title} con increibles ofertas online`}
                    />
                : <meta
                    name="description"
                    content="La comunidad de compra y venta online más grande de América Latina."
                    />
            }
        </Helmet>
    )
}
