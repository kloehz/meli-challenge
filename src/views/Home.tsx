import { Seo } from "../components/seo/Seo"

export const Home = () => {
    return (
        <>
            <Seo title={'Inicio'} />
            <div className="home-container">
                <img src='https://http2.mlstatic.com/frontend-assets/ui-navigation/5.16.1/mercadolibre/logo__large_plus.png' />
                <div className="home-title-container">
                    <h1>Challenge para</h1>
                    <h1 className="home-meli-word">&nbsp;MercadoLibre</h1>
                </div>
                <h2>De Guido Leonel Cotelesso</h2>
            </div>
        </>
    )
}