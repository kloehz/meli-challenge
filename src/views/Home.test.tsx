import { shallow } from "enzyme"
import { Seo } from "../components/seo/Seo";
import { Home } from "./Home"

describe('Home screen Tests', () => {
    
    const wrapper = shallow(<Home />);

    test('should be match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be display the correct image', () => {
        expect(wrapper.find('img').prop('src')).toBe('https://http2.mlstatic.com/frontend-assets/ui-navigation/5.16.1/mercadolibre/logo__large_plus.png');
    });

    test('SEO component should be rendered', () => {
        expect(wrapper.containsMatchingElement(<Seo />)).toEqual(true);
    });
    
})
