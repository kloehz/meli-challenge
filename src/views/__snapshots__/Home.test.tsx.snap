// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen Tests should be match snapshot 1`] = `ShallowWrapper {}`;
